lib/temporal/testing/replay_tester.rb

```diff
@@ -0,0 +1,136 @@
+      #   can happen in certain versions of certain SDKs. Use the correct_event_types function below to convert these.
+      #   See that function's comment for more details.
+      def replay_history_json_file(workflow_class, path)
+        json = File.read(path)
+        replay_history_json(workflow_class, json)
+      end
+
+      # Runs a replay test on a JSON string directly. See comment on replay_history_json_file more details.
+      def replay_history_json(workflow_class, json)
+        raw_history = Temporalio::Api::History::V1::History.decode_json(json, ignore_unknown_fields: true)
+        replay_history(workflow_class, Workflow::History.new(raw_history.events))
+      end
+
+      # Runs a replay test by loading a file from the given file path that contains protobuf binary. This can
+      # be downloaded usin the .get_workflow_history_protobuf method on the Temporal client or possibly through
+      # the use of third party tools like grpcurl.
+      #
+      # If the replay test succeeds, the method will return silently. If the replay tests fails, an error will be raised.
+      #
+      # IMPORTANT: This file is a binary file, not a text file. The protobuf obtained from get_workflow_history_protobuf
+      # must be written using binary file options.
+      def replay_history_protobuf_file(workflow_class, path)
+        protobuf = File.open(path, 'rb', &:read)
+        replay_history_protobuf(workflow_class, protobuf)
+      end
+
+      # Runs a replay test binary using a protobuf string
+      #
+      # If the replay test succeeds, the method will return silently. If the replay tests fails, an error will be raised.
+      def replay_history_protobuf(workflow_class, proto)
+        raw_history = Temporalio::Api::History::V1::History.decode(proto)
+        replay_history(workflow_class, Workflow::History.new(raw_history.events))
+      end
+
+      # Runs a replay test by using the specific Temporal::Workflow::History object. This can only be obtained by
+      # calling the .get_workflow_history method on the Temporal client to download a history from Temporal server
+      # into memory.
+      #
+      # If the replay test succeeds, the method will return silently. If the replay tests fails, an error will be raised.
+      def replay_history(workflow_class, history)
+        # This code roughly resembles the workflow TaskProcessor but with history being fed in rather
+        # than being pulled via a workflow task, no query support, no metrics, and other
+        # simplifications. Fake metadata needs to be provided.
+        metadata = Temporal::Metadata::WorkflowTask.new(
+          namespace: 'replay-test',
+          id: 1,
+          task_token: '',
+          attempt: 1,
+          workflow_run_id: 'run_id',
+          workflow_id: 'workflow_id',
+          workflow_name: history.find_event_by_id(1).attributes.workflow_type.name
```

Comment:
nit: Could probably use a comment or assertion that this is the event you expect.